modules/shortcodes/instagram.php

```diff
@@ -206,154 +185,6 @@ function jetpack_instagram_get_access_token() {
- *
- * @param string $url The URL that was attempted to embed.
- *
- * @return string The linked URL to the Instagram item.
- */
-function jetpack_instagram_output_errored_embed( $url ) {
-	return sprintf(
-		'<a href="%s">%s</a>',
-		esc_url( $url ),
-		esc_url_raw( $url )
-	);
-}
-
-/**
- * Should the request to fetch embed information be cached?
- *
- * @param array  $matches    Array of matches from the regex.
- * @param array  $atts       The original unmodified attributes.
- * @param string $passed_url The original URL that was matched by the regex.
- *
- * @return bool
- */
-function jetpack_instagram_use_cache( $matches, $atts, $passed_url ) {
-	/**
-	 * Filter Object Caching for response from Instagram.
-	 *
-	 * Allow enabling of object caching for the response sent by Instagram when querying for Instagram image HTML.
-	 *
-	 * @module shortcodes
-	 *
-	 * @since  3.3.0
-	 *
-	 * @param        bool        false Object caching is off by default.
-	 * @param array  $matches    Array of Instagram URLs found in the post.
-	 * @param array  $atts       Instagram Shortcode attributes.
-	 * @param string $passed_url Instagram API URL.
-	 */
-	return apply_filters( 'instagram_cache_oembed_api_response_body', false, $matches, $atts, $passed_url );
-}
-
-/**
- * Handles the logic for actually fetching information for an Instagram embed, conditionally proxying the
- * request through the WordPress.com API if this is not a WordPress.com site.
- *
- * @param array $args An array of arguments to pass to the embed API.
- *
- * @return mixed An object if successful or a WP_Error object
- */
-function jetpack_instagram_fetch_embed( $args ) {
-	$access_token = jetpack_instagram_get_access_token();
-
-	// Attempt to clean query params from the URL since Facebook's new API for Instagram
-	// embeds does not like query parameters. See p7H4VZ-2DU-p2.
-	$parsed_url = wp_parse_url( $args['url'] );
-	if ( $parsed_url ) {
-		$args['url'] = 'https://www.instagram.com' . $parsed_url['path'];
-	}
-
-	// If an access token exists, which will be the case for WPCOM, then we will call the Facebook API directly.
-	// Otherwise, proxy the request through the WordPress.com API using the blog token to sign the request.
-	if ( ! empty( $access_token ) ) {
-		$args['access_token'] = $access_token;
-		$url                  = esc_url_raw(
-			add_query_arg(
-				$args,
-				'https://graph.facebook.com/v5.0/instagram_oembed/'
-			)
-		);
-		$response             = wp_remote_get( $url, array( 'redirection' => 0 ) );
-
-		// Unset before calling the action below.
-		unset( $args['access_token'] );
-
-		/**
-		 * Fires after making a request for an Instagram embed.
-		 *
-		 * @module shortcodes
-		 *
-		 * @since  9.1.0
-		 *
-		 * @param array $response The response from the embed request.
-		 * @param array $url      The arguments sent with the request.
-		 */
-		do_action( 'jetpack_instagram_embed_request', $response, $args );
```

Comment:
With this change and the removal of the action, we lose a way to keep track of the requests. Is that alright?

If it is, we should add that action to the list of deprecated hooks alongside `instagram_cache_oembed_api_response_body `.

Reply:
Ohh, I think I missed that when rebasing. Thanks for raising this! I think we want to keep it. I'll add it back.

Reply:
Took me a bit to find an action that we can tap into. The best I could find is fairly generic and bit ugly (`http_api_debug`) -- I couldn't find anything more oEmbed specific that retained the errors (they are almost always discarded, and `false` is returned instead).

I think we have two options:
1. Do nothing here; instead, change the action added in D50894-code to hook into `http_api_debug` (and bail early if the `$url` doesn't start with `https://graph.facebook.com/v5.0/instagram_oembed/`) instead of `jetpack_instagram_embed_request`, and discard the latter (which conveniently hasn't been part of any public JP release yet).
2. Call the `jetpack_instagram_embed_request` action from the `http_api_debug` action:

```diff
diff --git a/modules/shortcodes/instagram.php b/modules/shortcodes/instagram.php
index 8b2ba5cfc..973057c11 100644
--- a/modules/shortcodes/instagram.php
+++ b/modules/shortcodes/instagram.php
@@ -164,6 +164,37 @@ function jetpack_instagram_oembed_remote_get_args( $args, $url ) {
 }
 add_filter( 'oembed_remote_get_args', 'jetpack_instagram_oembed_remote_get_args', 10, 2 );
 
+/**
+ * Call the `jetpack_instagram_embed_request` action upon HTTP response from the IG oEmbed API.
+ *
+ * @param array|WP_Error $response    HTTP response or WP_Error object.
+ * @param string         $context     Context under which the hook is fired.
+ * @param string         $class       HTTP transport used.
+ * @param array          $args        HTTP request arguments.
+ * @param string         $url         The request URL.
+ */
+function jetpack_instagram_http_api_debug( $response, $context, $class, $args, $url ) {
+	if ( ! wp_startswith( $url, 'https://graph.facebook.com/v5.0/instagram_oembed/' ) ) {
+		return;
+	}
+
+	// Unset before calling the action below.
+	unset( $args['access_token'] );
+
+	/**
+	 * Fires after making a request for an Instagram embed.
+	 *
+	 * @module shortcodes
+	 *
+	 * @since  9.1.0
+	 *
+	 * @param array $response The response from the embed request.
+	 * @param array $args     The arguments sent with the request.
+	 */
+	do_action( 'jetpack_instagram_embed_request', $response, $args );
+}
+add_action( 'http_api_debug', 'jetpack_instagram_http_api_debug', 10, 5 );
+
 /**
  * Fetches a Facebook API access token used for query for Instagram embed information, if one is set.
  *
```

Any preference? :thinking:

Reply:
I'd personally vote for option 1.

Reply:
Thanks! I'll file a diff.

Reply:
D51050-code